docs/monitors/haproxy.md

```diff
@@ -0,0 +1,207 @@
+Define the HAProxy monitor in the agent configuration file and provide the `csv export` URL where HAProxy
+stats are served in CSV format. This monitor supports basic HTTP authentication. Simply provide the
+username and password if required as shown below:
+```
+...
+monitors:
+  - type: haproxy
+    url: "http://localhost:8404/stats?stats;csv"
+    username: "your username here"
+    password: "your password here"
+...
+```
+Note: Only stats pertaining to the configured proxies (i.e. listeners, frontends, backends, and servers)
+are available from an HTTP endpoint. HAProxy process level stats given by command [show info](https://cbonte.github.io/haproxy-dconv/1.8/management.html#9.3-show%20info)
+are not available. The `show info` stats can only be scraped from a configured stats UNIX socket.
+
+### UNIX Socket Config
+The location of the HAProxy socket file is defined in the HAProxy config file, as in the following example:
+```
+global
+    daemon
+    stats socket /var/run/haproxy.sock
+    stats timeout 2m
+```
+For the given location of the socket file, configure the HAProxy monitor URL as shown below:
+```
+...
+monitors:
+  - type: haproxy
+    url: "unix:/var/run/haproxy.sock"
+...
+```
+Note: The agent process needs read/write permissions to the HAProxy socket file.
+
+
+## Configuration
+
+To activate this monitor in the Smart Agent, add the following to your
+agent config:
+
+```
+monitors:  # All monitor config goes under this key
+ - type: haproxy
+   ...  # Additional config
+```
+
+**For a list of monitor options that are common to all monitors, see [Common
+Configuration](../monitor-config.md#common-configuration).**
+
+
+| Config option | Required | Type | Description |
+| --- | --- | --- | --- |
+| `username` | no | `string` | Basic Auth username to use on each request, if any. |
+| `password` | no | `string` | Basic Auth password to use on each request, if any. |
+| `url` | **yes** | `string` | URL on which to scrape HAProxy. The URL scheme `http://` is designated as http-type configuration while `unix://` socket-type. |
+| `sslVerify` | no | `bool` | Flag that enables SSL certificate verification for the scrape URI. (**default:** `true`) |
+| `timeout` | no | `int64` | Timeout for trying to get stats from HAProxy. This should be a duration string that is accepted by https://golang.org/pkg/time/#ParseDuration (**default:** `5s`) |
+| `proxies` | no | `list of strings` | A list of the pxname(s) and svname(s) to monitor (e.g. `["http-in", "server1", "backend"]`). If empty then metrics of all proxies will be reported. |
+
+
+## Metrics
+
+These are the metrics available for this monitor.
+Metrics that are categorized as
+[container/host](https://docs.signalfx.com/en/latest/admin-guide/usage.html#about-custom-bundled-and-high-resolution-metrics)
+(*default*) are ***in bold and italics*** in the list below.
+
+
+ - `counter.connection_total` (*cumulative*)<br>    Cumulative number of connections. Values reported for frontends.
```

Comment:
If this is our own plugin now (not the collectd one), then make the metric names cleaner (i.e. no counter/derive prefix, no abbreviations of things like `abrt` for `abort`, `byp` for `bypassed`, etc.

Reply:
ok